devtools/etdump/data_sink.h

```diff
@@ -0,0 +1,28 @@
+// (c) Meta Platforms, Inc. and affiliates. Confidential and proprietary.
+
+#pragma once
+
+#include <executorch/devtools/etdump/data_sink_base.h>
+#include <executorch/runtime/core/exec_aten/exec_aten.h>
+#include <executorch/runtime/core/span.h>
+
+namespace executorch {
+namespace etdump {
+
+class DataSink : public DataSinkBase {
```

Comment:
Since there will be multiple subclasses of DataSinkBase, it's important to pick names that differentiate them: "DataSink" is too generic. This could be "BufferDataSink" since it writes into a buffer.